README.md

```diff
@@ -122,42 +122,62 @@ The particle can be accesses using `iter->` (`*iter` is also possible).
-1. blub
-2. blob
-3. blab
-
-
-#### When it is necessary
-You have to update the container when the two conditions are fullfilled:
-* If you moved particles
-* You want to use `iteratePairwise()` or a RegionParticleIterator
-
-#### When it is not enough
-If you moved particles by more than one interaction length.
-If you are planning to move particles by a long distance,
-e.g. because of boundary conditions please delete the particles and add them again:
+### Simulation Loop
+One simulation loop should always consist of the following phases:
+
+1. Updating the Container, which returns a vector of all invalid == leaving particles!
+   ```C++
+   auto invalidParticles = autoPas.updateContainer();
+   ```
+
+2. Handling the leaving particles
+   * Apply boundary conditions on them
+   
+   * Potentially send them to other mpi-processes
+   
+   * Add them to the containers using
+      ```C++
+      addParticle(particle)
+      ```
+
+3. Handle halo particles:
+   * Identify the halo particles by use of AutoPas' iterators and send them in a similar way as the leaving particles.
+
+   * Add the particles as haloParticles using 
+      ```C++
+      addOrUpdateHaloParticle(haloParticle)
+      ```
+
+4. Perform an iteratePairwise step.
+   ```C++
+   autoPas.iteratePairwise(functor);
+   ```
+
+In some iterations step 1. will return an empty list of invalid particles to benefit of not rebuilding the containers and the associated neighbor lists.
+
+### Using multiple functors
+
+AutoPas is able to work with simulation setups using multiple functors.
+A good example for that is the sph example found under examples/sph or examples/sph-mpi.
+There exist some things you have to be careful about when using multiple functors:
+* If you use multiple functors it is necessary that all functors support the same newton3 options. If there is one functor not supporting newton3, you have to disable newton3 support for AutoPas by calling
+  ```C++
+  autoPas.setAllowedNewton3Options({false});
+  ```
+
+* If you have `n` functors within one iteration and update the particle position only at the end or start of the iteration, the rebuildFrequency and the samplingRate have to be a multiple of `n`.   
+
+### Inserting additional particles
+When inserting additional particles, you always have to enforce a containerUpdate before doing that on ALL AutoPas instances, i.e., on all mpi processes, by calling  
 ```C++
-std::vector<autopas::sph::SPHParticle> invalidParticles;
-for (auto part = sphSystem.begin(); part.isValid(); ++part) {
-  if (/*check*/) {
-    invalidParticles.push_back(*part);
-    part.deleteCurrentParticle();
-  }
-}
-for (auto p: invalidParticles) {
-  sphSystem.addParticle(p);
-}
-```
-
-#### Special exceptions
-* Verlet-Lists, here it is safe to not update the container as long as particles move not more than a skin radius.
+autoPas.updateContainerForced();
+``` 
+This will invalidate the internal neighbor lists and containers.
 
```

Comment:
Maybe we should describe this ownership model for particles in a dedicated section (can happen in a separate PR)

Reply:
ok

Reply:
#281